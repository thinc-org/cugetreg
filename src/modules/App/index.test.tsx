import { shallow } from 'enzyme'

describe('App module', () => {
  const mockRouter = jest.fn()
  const mockUseApp = jest.fn()
  const mockPageProps = { test: 'test' }
  const mockComponent = () => <div></div>
  const mockClient = jest.fn()
  jest.doMock('./hooks/useApp', () => ({ useApp: mockUseApp }))
  jest.doMock('@/services/apollo', () => ({ client: mockClient }))

  const DEFAULT_ENV = process.env

  beforeEach(() => {
    jest.resetModules()
    process.env = { ...DEFAULT_ENV }
  })

  afterAll(() => {
    process.env = DEFAULT_ENV
  })

  it.each`
    forceDark | environment
    ${true}   | ${'dev'}
    ${true}   | ${'beta'}
    ${true}   | ${'production'}
    ${false}  | ${'dev'}
    ${false}  | ${'beta'}
    ${false}  | ${'production'}
  `('Should app component render preperly when forceDark=$forceDark', async ({ forceDark, environment }) => {
    process.env.NEXT_PUBLIC_ENVIRONMENT = environment

    const { App } = await import('.')
    const { AppProvider } = await import('@/modules/App/context')
    const { DefaultSeo } = await import('next-seo')
    const wrapper = shallow(
      <App forceDark={forceDark} Component={mockComponent} router={mockRouter as any} pageProps={mockPageProps} />
    )
    expect(mockUseApp).toBeCalledWith(mockRouter)
    expect(wrapper.find(mockComponent).props()).toEqual(mockPageProps)
    expect(wrapper.find(AppProvider).prop('forceDark')).toBe(false)
    expect(wrapper.find(DefaultSeo).prop('dangerouslySetAllPagesToNoFollow')).toBe(environment !== 'production')
    expect(wrapper.find(DefaultSeo).prop('dangerouslySetAllPagesToNoIndex')).toBe(environment !== 'production')
    expect(wrapper).toMatchSnapshot()
  })
})
