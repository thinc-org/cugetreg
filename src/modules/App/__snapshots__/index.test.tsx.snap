// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App module Should app component render preperly when forceDark=false 1`] = `
<AppProvider
  emotionCache={
    Object {
      "insert": [Function],
      "inserted": Object {},
      "key": "css",
      "nonce": undefined,
      "registered": Object {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}@keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}@keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}@keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}@keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}.go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": Array [],
      },
    }
  }
  forceDark={false}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        Array [
          Object {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          Object {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={true}
      dangerouslySetAllPagesToNoIndex={true}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        Object {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": Array [
            Object {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:3000/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:3000",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module Should app component render preperly when forceDark=false 2`] = `
<AppProvider
  emotionCache={
    Object {
      "insert": [Function],
      "inserted": Object {},
      "key": "css",
      "nonce": undefined,
      "registered": Object {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}@keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}@keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}@keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}@keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}.go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": Array [],
      },
    }
  }
  forceDark={false}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        Array [
          Object {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          Object {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={true}
      dangerouslySetAllPagesToNoIndex={true}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        Object {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": Array [
            Object {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:3000/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:3000",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module Should app component render preperly when forceDark=false 3`] = `
<AppProvider
  emotionCache={
    Object {
      "insert": [Function],
      "inserted": Object {},
      "key": "css",
      "nonce": undefined,
      "registered": Object {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}@keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}@keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}@keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}@keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}.go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": Array [],
      },
    }
  }
  forceDark={false}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        Array [
          Object {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          Object {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={false}
      dangerouslySetAllPagesToNoIndex={false}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        Object {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": Array [
            Object {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:3000/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:3000",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module Should app component render preperly when forceDark=true 1`] = `
<AppProvider
  emotionCache={
    Object {
      "insert": [Function],
      "inserted": Object {},
      "key": "css",
      "nonce": undefined,
      "registered": Object {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}@keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}@keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}@keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}@keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}.go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": Array [],
      },
    }
  }
  forceDark={true}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        Array [
          Object {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          Object {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={true}
      dangerouslySetAllPagesToNoIndex={true}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        Object {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": Array [
            Object {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:3000/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:3000",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module Should app component render preperly when forceDark=true 2`] = `
<AppProvider
  emotionCache={
    Object {
      "insert": [Function],
      "inserted": Object {},
      "key": "css",
      "nonce": undefined,
      "registered": Object {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}@keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}@keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}@keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}@keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}.go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": Array [],
      },
    }
  }
  forceDark={true}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        Array [
          Object {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          Object {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={true}
      dangerouslySetAllPagesToNoIndex={true}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        Object {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": Array [
            Object {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:3000/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:3000",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module Should app component render preperly when forceDark=true 3`] = `
<AppProvider
  emotionCache={
    Object {
      "insert": [Function],
      "inserted": Object {},
      "key": "css",
      "nonce": undefined,
      "registered": Object {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}@keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}@keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}@keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}@keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}@keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}.go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": Array [],
      },
    }
  }
  forceDark={true}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        Array [
          Object {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          Object {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={false}
      dangerouslySetAllPagesToNoIndex={false}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        Object {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": Array [
            Object {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:3000/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:3000",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
  </TrackPageChange>
</AppProvider>
`;
